primefaces/src/main/java/org/primefaces/util/ComponentUtils.java

```diff
@@ -698,7 +710,9 @@ public static <T> T executeInRequestScope(FacesContext context, String var, Obje
 
         Object oldValue = requestMap.remove(var);
         try {
-            requestMap.put(var, value);
+            if (value != null) {
+                requestMap.put(var, value);
+            }
 
             return callback.get();
```

Comment:
Simpler this way no?

```java
Object oldValue = requestMap.put(var, value);
try {
    return callback.get();
}
finally {
    if (oldValue != null) {
        requestMap.put(var, oldValue);
    }
}
```

Reply:
I think you are mixing up the two methods. The execute in Request Scope requires a return value while executeStateless does not its a void.

Reply:
For example i need to remove `Constants.HELPER_RENDERER` from the state map, run some code, and then re-instate the `Constants.HELPER_RENDERER` value.

Reply:
You sure? Don't think I'm mixing anything here, also why having this null check before `requestMap.put(var, value);`?

Reply:
My coment is on executeInRequestScope not executeInStatelessScope 😄

Reply:
Resolved